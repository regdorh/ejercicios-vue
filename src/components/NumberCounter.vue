<template>
    <div class="wrapper">
        <h1>
            Contador
        </h1>

        <p>{{ count }}</p>

        <div class="botones">
            <button class="botones-increment" @click="increment">Incrementar</button>
            <button 
                @click="decrement"
                class="botones-decrement"
                :class="{
                    deshabilitado: count === 0
                }"
                :disabled="count === 0 ? true: false"
            >Decrementar</button>
        </div>
    </div>
</template>

<script>
import { mapActions, mapGetters } from "vuex";

export default {
    name: "NumberCounter",
    computed: {
        ...mapGetters({
            count: 'getCount'
        })
    },
    methods: {
        ...mapActions({
            increment: 'increment',
            decrement: 'decrement'
        })
    }
}

</script>


<style scoped>
    .botones {
        display: flex;
        gap: 2rem;
        justify-content: center;
    }
    p {
        font-size: 5rem;
    }
    .botones-increment {
        background-color: rgb(0, 81, 255);
        border-radius: 1rem;
        padding: 1rem 2rem;
        font-weight: 900;
        color: #FFF;
        cursor: pointer;
    }

    .botones-decrement {
        background-color: rgb(255, 0, 55);
        border-radius: 1rem;
        padding: 1rem 2rem;
        font-weight: 900;
        color: #FFF;
        cursor: pointer;
    }
    .deshabilitado {
        opacity: .5;
        cursor: not-allowed;
    }
</style>