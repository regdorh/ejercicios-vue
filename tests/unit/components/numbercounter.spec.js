import { shallowMount } from "@vue/test-utils"; 
import NumberCounter from "@/components/NumberCounter.vue";
 
const store = {
  state: { 
    count: 0, 
  },
  getters: { 
    getCount(state) {
      return state.count; 
    },
  },
  mutations: { 
    INCREMENT(state) { 
      state.count++;
    },
    DECREMENT(state) { 
      state.count > 0 && state.count--;
    },
  },
  actions: { 
    async increment(context) { 
      context.commit("INCREMENT"); 
    },
    async decrement(context) { 
      context.commit("DECREMENT");
    },
  },
  modules: {},
};

describe("Componente NumberCounter.vue", () => {

  test("Validación de match con el snapshot (Coincidir con la instantánea)", () => {
    
    const wrapper = shallowMount(NumberCounter, {
      global: {
        mocks: { 
          $store: store,
        },
      },
    });

    expect(wrapper.html()).toMatchSnapshot(); 
  });
});