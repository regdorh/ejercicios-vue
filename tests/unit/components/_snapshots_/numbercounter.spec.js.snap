exports[`Componente NumberCounter.vue Validación de match con el snapshot (Coincidir con la instantánea) 1`] = `
<div class="wrapper">
  <h1> Contador </h1>
  <p>getCount(state) {
    return state.count;
    }</p>
  <div class="botones"><button class="botones-increment">Incrementar</button><button class="botones-decrement">Decrementar</button></div>
</div>
`;